core/bones/relational.py

```diff
@@ -443,8 +443,10 @@ def isInvalid(self, key):
     def parseSubfieldsFromClient(self):
         return self.using is not None
 
-    def singleValueFromClient(self, value, skel, name, origData):
-        oldValues = skel[name]
+    def singleValueFromClient(self, value: Any, skel: 'SkeletonInstance',
+                              bone_name: str, client_data: dict
+                              ) -> tuple[Any, list[ReadFromClientError] | None]:
```

Comment:
Is it necessary to provide full type annotations in sub-classes as well?
I would like to keep maintenance overhead low, if possible.